### PR TITLE
Add lint configurations and fix lint issues

### PR DESCRIPTION
**Pull request by zengyan-amazon**
*Description of changes:*

Add following lint related config files:
* `.eslintrc.js`
* `.eslintignore`
* `.sass-lint.yml`

to execute lint run following command from plugin root directory:
`yarn lint`
or only run lint against typescript/javascript code only or scss only:
```
yarn lint:es
```
or
```
yarn lint:sass
```

to get CLI options for eslint, run:
```
yarn lint:es --help
```

**Testing**
lint output for server side code so far:
```
% yarn lint:es --ignore-pattern "public/**/*"
yarn run v1.22.0
$ node ../../scripts/eslint --ignore-pattern 'public/**/*'

/Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/build_tools/build_plugin.js
  1:1  error  File must start with a license header  @kbn/eslint/require-license-header

/Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/server/auth/types/basic/basic_auth.ts
   16:1  error  File has too many classes (2). Maximum allowed is 1  max-classes-per-file
  165:7  error  Unexpected console statement                         no-console

/Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/server/backend/opendistro_security_configuration_plugin.ts
  16:1  error  Prefer named exports  import/no-default-export

/Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/server/backend/opendistro_security_plugin.ts
  16:1  error  Prefer named exports  import/no-default-export

/Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/server/plugin.ts
  32:8  error  Identifier 'opendistro_security_configuratoin_plugin' is not in camel case  @typescript-eslint/camelcase
  67:9  error  Identifier 'opendistro_security_configuratoin_plugin' is not in camel case  @typescript-eslint/camelcase

/Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/server/routes/test_routes.ts
  67:9  error  Unexpected console statement  no-console

✖ 8 problems (8 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+`
```
Comment (seraphjiang) [inline .eslintrc.js:16]: Is this ` expected?

File: .eslintrc.js [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
 
 module.exports = {	
   root: true,	
   extends: ['@elastic/eslint-config-kibana', 'plugin:@elastic/eui/recommended'],
   rules: {
-    "@kbn/eslint/require-license-header": "off"
-  }
+    // "@kbn/eslint/require-license-header": "off"
```
Comment (seraphjiang) [inline .eslintrc.js:22]: Should we remove or replace with opendiatro one

File: .eslintrc.js (hunk 1 above)
Comment (zengyan-amazon) [reply .eslintrc.js:16]: the ` works as multiline quotes, to make the license header a multi line string to lint when checking

Comment (seraphjiang) [reply .eslintrc.js:16]: Thanks,?I didn’t catch that

File: .eslintrc.js (hunk 2 above)
Comment (zengyan-amazon) [reply .eslintrc.js:22]: the rules are defined in Kibana repo, we are switching to use their rules to check our license header. Can remove the `// "@kbn/eslint/require-license-header": "off"` here

**Pull request merged**